pulsar-transaction/buffer/src/main/java/org/apache/pulsar/transaction/buffer/impl/InMemTransactionBuffer.java

```diff
@@ -79,6 +81,41 @@ public int numEntries() {
             }
         }
 
+        @Override
+        public long committedAtLedgerId() {
+            return committedAtLedgerId;
+        }
+
+        @Override
+        public long committedAtEntryId() {
+            return committedAtEntryId;
+        }
+
+        @Override
+        public long lastSequenceId() {
+            return entries.lastKey();
+        }
+
+        @Override
+        public CompletableFuture<SortedMap<Long, Position>> readEntries(int num, long startSequenceId) {
+            return FutureUtil.failedFuture(new UnsupportedOperationException());
```

Comment:
It is trivial to implement this method. So please implement it.

Reply:
Because it related to the entry position and there is no position in memory. So I think it is no usage in this implementation, right?